### ISSUE TITLE
September Update. Empty VM

### ISSUE THREAD

**Issue by dkatson** [Microsoft/AL#633]
Hi team, 
I tried to create September VM from template http://aka.ms/navdeveloperpreview and it was created as standard Windows VM, without NAV, VS Code and any Demo folders in C:\.

I tried with US

**Comment by Gallimathias** [comment]
NAV is hosted in Docker and has in the current update only the web client.

Just run "C: \DEMO\SetupDesktop. ps1".

And look at the landing page, it's all there.

[Team Blog](https://blogs.msdn.microsoft.com/nav/2017/09/07/nav-development-preview-september-update/)

**Comment by dkatson** [comment]
Hi Marcus,

**Comment by Gallimathias** [comment]
Hi Dmitry  :D

I think you lost your message. ^^

**Comment by dkatson** [comment]
Hi Marcus,
That is the question. After creation of VM I don't see any landing page, and also "C:\DEMO" is missed.

**Comment by dkatson** [comment]
Oh, I found this in Azure log
`OPERATION ID
48AB7AB9BFF6E7C3
TRACKING ID
03a4dc1b-6cf7-4f97-bccd-4d430f917e42
STATUS
BadRequest
PROVISIONING STATE
Failed
TIMESTAMP
9/7/2017, 1:15:44 PM
DURATION
0.6577485 seconds
TYPE
Microsoft.Compute/virtualMachines/extensions
RESOURCE ID
/subscriptions/d0e75c3c-a34a-455b-9d44-667cd10c9756/resourceGroups/navdesign/providers/Microsoft.Compute/virtualMachines/ndnav2018prvus/extensions/vmextension1
STATUSMESSAGE
{
  "error": {
    "code": "InvalidTemplate",
    "message": "Unable to process template language expressions for resource '/subscriptions/d0e75c3c-a34a-455b-9d44-667cd10c9756/resourceGroups/navdesign/providers/Microsoft.Compute/virtualMachines/ndnav2018prvus/extensions/vmextension1' at line '1' and column '7389'. 'The language expression property 'templateLink' doesn't exist, available properties are 'template, parameters, mode, debugSetting, provisioningState, validationLevel'.'"
  }
}
RESOURCE
ndnav2018prvus/vmextension1`

**Comment by Gallimathias** [comment]
Did you take the link here from the readme or from the blog or did you create the VM using the gallery?

Have you already entered your VM domain in your browser?

**Comment by dkatson** [comment]
I used a link from the blog. http://aka.ms/navdeveloperpreview 
I suppose the problem is here 
![image](https://user-images.githubusercontent.com/29101017/30159760-67f0e0ae-93d2-11e7-99c1-6dcedc1e9821.png)

because after creation of VM it fails on running initialize.ps1

**Comment by Gallimathias** [comment]
try changing that to template. However, it's strange, I had the script without any problems.

**Comment by dkatson** [comment]
I tried... if I change to "properties.template.uri" then it says that there is no "uri" as a parameter

**Comment by Gallimathias** [comment]
I missing the dependensOn Prop on your file

´´´´json
"apiVersion": "2015-06-15",
            "type": "Microsoft.Compute/virtualMachines/extensions",
            "name": "[concat(variables('server1Name'),'/vmextension1')]",
            "location": "[resourceGroup().location]",
            "tags": {
                "displayName": "PowerShellScript2"
            },
            "dependsOn": [
                "[concat('Microsoft.Compute/virtualMachines/', variables('server1Name'))]"
            ],
            "properties": {
                "publisher": "Microsoft.Compute",
                "type": "CustomScriptExtension",
                "typeHandlerVersion": "1.4",
                "autoUpgradeMinorVersion": true,
                "settings": {
                    "fileUris": [
                        "[uri(deployment().properties.templateLink.uri, 'initialize.ps1')]"
                    ],
´´´´

**Comment by dkatson** [comment]
Strange, I changed resource group to new, created new VM and it created as expected.
Thanks.

**Comment by Gallimathias** [comment]
Heey no problem for that there's community ^^

**Comment by hgnamvn** [comment]
Hi !

I also get this issue.
1. I create VM from gallery. But the version still is August Update (10.0.18031). I create that VM after I reed this post.
2. I create VM from the link:  http://aka.ms/navdeveloperpreview I select W1 version. the VM is create OK but there aren't any NAV, VS Code. The DEMO folder only exist the status file:

_<font color="Gray">2:03:35 AM Starting initialization</font>
<font color="Gray">2:03:35 AM TemplateLink: https://raw.githubusercontent.com/NAVDEMO/DOCKER/master/navdeveloperpreview.json</font>
<font color="Gray">2:03:35 AM Upgrading Docker Engine</font>
<font color="Gray">2:04:01 AM Logging in to navdocker.azurecr.io</font>
<font color="Gray">2:04:04 AM pulling microsoft/windowsservercore</font>
<font color="Gray">2:04:07 AM pulling navdocker.azurecr.io/dynamics-nav:devpreview</font>
<font color="Gray">2:31:44 AM VM already initialized.</font>_

**Now:** I will try to do it again.

**Comment by hgnamvn** [comment]
Sorry, I understood my issue.
- Because I created a VM from Step 1, when more VM at step 2,  I key the VM's name is the same with VM's name at step 1.
- After I have deleted all VM, I create new VM again from template. It work fine.

However, issue of step 1 still exist. Why do I create from Gallery, The version still is August Update (10.0.18031).

**Comment by freddydk** [comment]
A few things.
In Azure template deployment, if you select Edit Template to see the template, then the templatelink will become null and you cannot deploy. This is a behavior of Azure Resource Manager and has always worked like that.
I have requested the image in the Azure Gallery to be deleted - we do no longer have an image in the gallery, we have a template. We will request that this template is added to the gallery.
I have updated a new script for September Preview late Sunday evening and fixed a few issues (like restart) and now you will also get a finsql shortcut on the desktop.
